engine/src/main/java/org/hibernate/search/cfg/spi/impl/ExtendedSearchConfiguration.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.cfg.spi.impl;
```

Comment:
I don't think it's right to place a class in `impl` if it's intended by external components/integrators. Couldn't you just control this via a property? That wouldn't require any public API/SPI at all. If it's for experimenting for now, I wouldn't mind if there wasn't a constant for it.